tests/testthat/test-read_redcap_tidy.R

```diff
@@ -254,3 +254,67 @@ test_that("get_fields_to_drop handles checkboxes", {
   )
 })
 
+test_that("read_redcap_tidy returns metadata", {
+  httptest::with_mock_api({
+    out <- read_redcap_tidy(redcap_uri, longitudinal_token)
+  })
+
+  expected_cols <- c(
+    "redcap_form_name", "redcap_form_label", "redcap_data", "redcap_metadata",
+    "redcap_events", "structure", "data_rows", "data_cols", "data_size",
+    "data_na_pct"
+  )
+
+  # metadata fields exist and correctly ordered
+  expect_equal(expected_cols, names(out))
+
+  # metadata fields have the correct data types
+
+  ## redcap_metadata and redcap_events fields consist of tibbles
+  c(out$redcap_metadata, out$redcap_events) %>%
```

Comment:
I thought I left a comment but somehow it disappeared. Can we do this one without `map` functions? I'm thinking:

``` r
expect_s3_class(out$redcap_metadata, "tbl")
expect_s3_class(out$redcap_events, "tbl")
```

I know this is nitpicky but I really think that it's good to keep test code as easy to read as possible (especially for someone like myself who doesn't code every day!)

Reply:
No worries! If want to do away with the `map` I think we'd need to just test the first element:
```r
expect_s3_class(out$redcap_metadata[[1]], "tbl")
expect_s3_class(out$redcap_events[[1]], "tbl")
```

`out$redcap_metadata` itself isn't a `tibble` it's a `list` so you get:
```r
expect_s3_class(out$redcap_metadata, "tbl")
#> Error: out$redcap_metadata is not an S3 object
```

Or if we're feeling really fiesty:
```r
expect_s3_class(sample(out$redcap_metadata, 1), "tbl")
expect_s3_class(sample(out$redcap_data, 1), "tbl")
```

to get a random element of out$redcap_* each time

Reply:
Unfortunately the most succinct way to make this test that I could think of is the way you wrote with `map_lgl`. The `sample` method shores up the test with randomness, but I don't really like that you technically can't say a test is the same from one run to the next. Doesn't seem kosher to me.

I think the best bet is to do indexing, and since the full length of what you're testing is (I believe) 6 tables, you could just write 6 expectation functions. It's not pretty, but we've done it in the past and it's readable.

Reply:
Ah, I see the difference now! And I'm just realizing for the first time that when you have a list column you can actually have different object types in different rows of a `tibble`! 😨 

So the `map` approach is more conservative from that point of view but I think checking the first element is good enough. I think the chance that someone will introduce a code change that would cause you to have a `tibble` in row 1 and some kind of other object in a different row is negligible.

The `sample` approach would introduce non-deterministic behavior into the test suite. There is a place for doing this kind of testing, but this is not that place :)

Reply:
Switched things to just test the first element. I would be hard pressed to think of a case where we *actually* accidentally create a list column with mixed object types. Although yes technically possible. Testing the first element seems good enough.